app/tests/e2e.test.ts

```diff
@@ -0,0 +1,115 @@
+import fetch from "node-fetch";
+const baseUrl = "http://localhost:5001/helloworld";
+const testId = "test-id-for-raml";
+const transformedTestId = "test-with-transformer";
+
+const httpRequest = async <T extends { message: string }>({
+  method,
+  content,
+  request = baseUrl,
+}: {
+  method: "post" | "put" | "get" | "patch" | "delete";
+  content?: string;
+  request?: string;
+}): Promise<{ status: number; message?: T["message"] }> => {
+  const response = await fetch(request, {
+    method,
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: content,
+  });
+
+  const body =
+    response.status === 204
+      ? response.statusText
```

Comment:
Yes, in case of a 204 with no body, the `statusText` returns with a string `No Content`. This is a unique behaviour for 204, and a reliable method to ensure that it is returning no content as expected.

Reply:
I've had a read from where `statusText` gets populated from. The `statusText` comes from the construction of a IncomingMessage inside the http/https library. From the documentation it says that if not supplied a message, then the default for the status code will be used. As the 204 code status message is the No Content [(mdn 204 sepc)](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204), means that the library never checks the body, it assumes from the status code.

A check then to prove that nothing is return would be to dump the buffer as proof.
```suggestion
      ? (await response.buffer()).length.toString()
```
With a change in the 204 tests to be 
```
expect(message).toBe(0);  
```
As toBe is a lossy check it won't matter that the message is a string.

Just one way of doing that check.

Reply:
Adopted this for the delete request as well as the head method.